src/main/java/com/google/sps/utilities/SearchDatastoreService.java

```diff
@@ -90,10 +96,61 @@
             
         }
 
+        switch(sortType.toLowerCase()) {
+            case "availability":
+                sortByAvailability(tutors);
+                break;
+            case "most-students":
+                sortByStudents(tutors, "most");
+                break;
+            case "least-students":
+                sortByStudents(tutors, "least");
+                break;
+            default:
+                //do nothing
+                break;
+        }
+
         return tutors;
         
     }
 
+    /**
+    * Sorts the list of tutors based on the number of students each tutor has. Sorts in descending order if 
+    * order = "most" or ascending order if order = "least". 
+    */
+    private void sortByStudents(ArrayList<Tutor> tutors, String order) {
+        StudentDatastoreService studentDatastore = new StudentDatastoreService();
+
+        for(Tutor tutor : tutors) {
+            tutor.setStudents(studentDatastore.getStudentsForTutor(tutor.getUserId()));
+        }
+
+        Collections.sort(tutors, new Comparator<Tutor>() {
+            @Override
+            public int compare(Tutor a, Tutor b) {
+                if(order.equals("most")) {
+                    return b.getStudents().size() - a.getStudents().size();
+                }
+
+                return a.getStudents().size() - b.getStudents().size();
+            }
```

Comment:
I don't know if got to look at this, but it could be cool to see what kind of sorting algorithms might be more efficient when making this comparison

Reply:
Yeah that could be cool. I think Collections.sort uses merge sort which is pretty good for longer lists.